core/cap-0035.md

```diff
@@ -254,22 +255,42 @@ index 7f08d757..e0585994 100644
 +{
 +    ClaimableBalanceID balanceID;
 +};
++
++/* SetTrustlineFlagsOp
++
++   Updates the flags of an existing trust line.
++   This is called by the issuer of the related asset.
++
++   Threshold: low
++
++   Result: SetTrustlineFlagsResult
++*/
++struct SetTrustlineFlagsOp

```

Comment:
❓  During the last open protocol meeting @stanford-scs raised the possibility of in the future there being an operation that allowed an account holder to add the clawback enabled flag on their own trustline. I'm trying to imagine what that change might look like if it was ever pursued given that this new operation is usable only by the issuer, and would likely remain usable only by the issue to preserve semantics.

If this idea was pursued in the future would we modify the semantics of SetTrustlineFlagsOp so that it could be used by either the issuer or the account holder with there being different rules about what flags could be set or cleared by different parties?

Or would we create a similar operation such as `SetOwnTrustlineFlagsOp` that had that other behavior?

Is there anything we could do to make it clearer that this operation is for issuers to use and not account holders? to set it apart from the get-go, and generally make its use clearer.

Reply:
This is a good question. One possibility is that we could include `AccountID trustor` in a union, with another possible value of  `AccountID issuer` in `SetTrustlineFlagsOp`. The union type will indicate if the operation is intended to act on a trustline as an issuer, or as its owner. In protocol 16, the owner case will always return malformed, but we can easily change that in future protocols.

Reply:
I changed `SetTrustLineFlagsOp` to store the account in a union, with only one type at the moment. The idea here is we can add a different type in the future to act on a trustline as the owner.

Reply:
Interesting. In this new model would the other future type be an AccountID that is the `issuer`? So issuer is source account and trustor is set, or account is source account and issuer is set?

This makes me wonder also about how this would be perceived alongside the existing ChangeTrustOp. Could flags be added to ChangeTrustOp in the future? I'm guessing no, although that would be conceptually elegant since it is already the operation for an account to mutate its own trustline.
https://github.com/stellar/stellar-core/blob/b09de37/src/xdr/Stellar-transaction.x#L212-L225

Reply:
I honestly don't feel great about this approach. Setting flags on my own trust line is conceptually quite different from the issuer setting flags on my trust line. I'd rather use different operations because
- it's more explicit about what is happening
- it permits variations:
    - should both versions have the same threshold? `ChangeTrust` is medium so I would expect that setting my own trust line flags would be medium too, whereas this operation is low
    - both versions can allow different flags, while keeping each implementation simple
- it doesn't prevent us from sharing parts of the underlying implementation, like we do for order book operations

What do you both think?

As for flags on `ChangeTrustOp`, no because (a) there is no extension for them and (b) because it would violate our principle of not extending operations. But in keeping with what I was saying in the previous paragraph, perhaps we should add a new operation that is like `SetOptions` but for trust lines that lets you do everything. Or perhaps not. But doing to separate operations preserves this flexibility, whereas one hybrid operation eliminates it immediately.

Reply:
I wanted to avoid the operational burden on adding a second operation, but your points are valid. I removed the union from `SetTrustLineFlagsOp`.